### ISSUE TITLE
DecLearn optimizer sample notebook gives poor accuracy

### ISSUE THREAD

**Issue by mvesin** [fedbiomed/fedbiomed#909]
When using `declearn-with-pytorch.ipynb` notebook with 1 node (keeping notebook parameters for rounds, batch_maxnum, test_ratio, etc.) we get a poor accuracy around 0.11 (which barely means a random answer for MNIST ?)

This might be:
- either a bug in Fed-BioMed / DecLearn integration ?
- or a need for tuning the parameters used for the DecLearn optimizer ? (would be nice for a demo notebook :-) )

Final output after `exp.run()` in `declearn-with-pytorch.ipynb`:

```
 -----------------------------------------------------------------
2023-11-10 14:40:23,440 fedbiomed DEBUG - Node: node_5153815b-cb76-49a8-9d5c-f0d955118869 polling for the tasks
2023-11-10 14:40:25,514 fedbiomed INFO - VALIDATION ON GLOBAL UPDATES 
					 NODE_ID: node_5153815b-cb76-49a8-9d5c-f0d955118869 
					 Round 3 | Iteration: 1/1 (100%) | Samples: 6000/6000
 					 ACCURACY: 0.112333 
					 ---------
```

As a comparison, running `101-getting_started.ipynb` notebook (which is basically the same but using a native pytorch optimizer), and just adding 3 lines for doing validation same as `declearn-with-pytorch.ipynb` we obtain around 0.95 accuracy:

Added to `101-getting-started.ipynb` notebook:

```
exp.set_test_ratio(0.1)
exp.set_test_on_local_updates(True)
exp.set_test_on_global_updates(True)
```

Final output after `exp.run()` in `101-getting-started.ipynb`:

```
 -----------------------------------------------------------------
2023-11-10 14:24:24,574 fedbiomed DEBUG - Node: node_5153815b-cb76-49a8-9d5c-f0d955118869 polling for the tasks
2023-11-10 14:24:26,635 fedbiomed INFO - VALIDATION ON GLOBAL UPDATES 
					 NODE_ID: node_5153815b-cb76-49a8-9d5c-f0d955118869 
					 Round 3 | Iteration: 1/1 (100%) | Samples: 6000/6000
 					 ACCURACY: 0.965667 
					 --------
```

**Comment by ybouilla** [comment]
The difference of accuracy is due to the fact learning rates are different from one notebook to another. Setting the same learning rate results in the same accuracy. I pushed a small PR for declearn-with-pytorch notebook that sets the same learning rate for each notebook. 

Closing the issue
